examples/chain_replication.pyv

```diff
@@ -0,0 +1,89 @@
+# node ids and states
+sort node_id
+sort node_state
+
+# ordering on node states
+immutable relation le(node_state, node_state)
+axiom le(X,X) # reflexive
+axiom le(X,Y) & le(Y,X) -> X = Y # anti-symmetric
+axiom le(X,Y) & le(Y,Z) -> le(X,Z) # transitive
+axiom le(X,Y) | le(Y,X) # total
+
+# empty state
+immutable constant empty: node_state
+
+# the empty state is the least state
+axiom le(empty, X)
+
+# let's say that there are three nodes
+immutable constant head: node_id
+immutable constant middle: node_id
+immutable constant tail: node_id
+
+# the head, middle, and tail nodes are all different
+# (this is too strong; it should be OK for them to all be the same, too)
+axiom head != middle & head != tail & middle != tail
+
+# nodes have states
+# TODO: maybe this could just be a big mutable relation
+mutable constant head_state : node_state
+mutable constant middle_state : node_state
+mutable constant tail_state : node_state
```

Comment:
I do think this probably could be a mutable function (or relation, but a function is easier) from nodes to states. That would remove the need for the axiom above, since if two nodes are equal, the function will give them only one state.

(Using a function would also lay the groundwork for supporting an arbitrary number of nodes in the chain someday.)

Reply:
OK, I'll try to make it a function.